File: bindings/generator/api/path.py
# Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS

from typing import Optional

from .common import EntryName, FsPath, Ref


def path_join(parent: Ref[FsPath], child: Ref[EntryName]) -> FsPath:
    raise NotImplementedError


def path_parent(path: Ref[FsPath]) -> FsPath:
    raise NotImplementedError


def path_filename(path: Ref[FsPath]) -> Optional[EntryName]:
    raise NotImplementedError


def path_split(path: Ref[FsPath]) -> list[EntryName]:
    raise NotImplementedError


def path_normalize(path: FsPath) -> FsPath:
    raise NotImplementedError

File: libparsec/src/path.rs
// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS

use libparsec_types::{EntryName, FsPath};

pub fn path_join(parent: &FsPath, child: &EntryName) -> FsPath {
    parent.join(child.clone())
}

pub fn path_parent(path: &FsPath) -> FsPath {
    path.parent()
}

pub fn path_filename(path: &FsPath) -> Option<&EntryName> {
    path.name()
}

pub fn path_split(path: &FsPath) -> Vec<EntryName> {
    path.parts().to_vec()
}

pub fn path_normalize(path: FsPath) -> FsPath {
    path.normalize()
}
